tests/drivers/test_raster_drivers.py

```diff
@@ -144,6 +146,22 @@ def test_lazy_loading(driver_path, provider, raster_file):
     assert all(np.all(data1[k] == data2[k]) for k in data1.keys())
 
 
+@pytest.mark.parametrize('provider', DRIVERS)
+def test_non_writable_lazy_loading(driver_path, provider, raster_file):
+    from terracotta import drivers
+    db = drivers.get_driver(driver_path, provider=provider)
+    keys = ('some', 'keynames')
+
+    db.create(keys)
+    db.insert(['some', 'value'], str(raster_file), skip_metadata=True)
+
+    # Manually set the meta store to un-writable
+    db.meta_store._WRITABLE = False
+
+    with pytest.raises(exceptions.DatabaseNotWritableError):
+        db.get_metadata(['some', 'value'])
```

Comment:
I think this will need a monkeypatch context, too.

Reply:
True, added it now.